src/Jaeger.Core/Reporters/RemoteReporter.cs

```diff
@@ -22,37 +23,56 @@ public class RemoteReporter : IReporter
         private readonly BufferBlock<ICommand> _commandQueue;
         private readonly Task _queueProcessorTask;
         private readonly TimeSpan _flushInterval;
-        private readonly Task _flushTask;
+        private readonly int _maxQueueSize;
         private readonly ISender _sender;
         private readonly IMetrics _metrics;
         private readonly ILogger _logger;
+        private System.Timers.Timer _timer;
 
         internal RemoteReporter(ISender sender, TimeSpan flushInterval, int maxQueueSize,
             IMetrics metrics, ILoggerFactory loggerFactory)
         {
             _sender = sender;
             _metrics = metrics;
             _logger = loggerFactory.CreateLogger<RemoteReporter>();
-            _commandQueue = new BufferBlock<ICommand>( new DataflowBlockOptions() {
+            _commandQueue = new BufferBlock<ICommand>(new DataflowBlockOptions()
+            {
                 BoundedCapacity = maxQueueSize
             });
 
             // start a thread to append spans
-            _queueProcessorTask = Task.Run(async () => { await ProcessQueueLoop(); });
-
+            _queueProcessorTask = Task.Run(ProcessQueueLoop);
             _flushInterval = flushInterval;
-            _flushTask = Task.Run(FlushLoop);
+            _maxQueueSize = maxQueueSize;
+            InitializeFlushTimer();
+        }
+
+        private void InitializeFlushTimer()
+        {
+            _timer = new System.Timers.Timer
+            {
+                AutoReset = false,
+                Interval = _flushInterval.TotalMilliseconds
+            };
+            _timer.Elapsed += FlushOnTimer;
+            _timer.Start();
         }
 
         public void Report(Span span)
         {
             // It's better to drop spans, than to block here
             var added = _commandQueue.Post(new AppendCommand(this, span));
-            
+
             if (!added)
             {
                 _metrics.ReporterDropped.Inc(1);
             }
+
+            if (_commandQueue.Count >= _maxQueueSize)
```

Comment:
This would defy the purpose of the queue size. We ONLY want to flush on the time limit. The queue size is to ensure there are NEVER more elements around. This includes the ones being flushed.